src/test/java/com/epam/test_generator/dao/SuitDAOTest.java

```diff
@@ -137,54 +137,33 @@ public void testUpdateName() {
     }
 
     @Test
-    public void testUpdateWithRowNumbers() {
+    public void testUpdateRowNumber() {
 
-        final List<Suit> expectedSuits = suitDAO.save(getExpectedSuitsFromDatabase());
-        final List<SuitRowNumberUpdate> sortedRowNumbers = getSortedSuitsRowNumbers(expectedSuits);
-
-        final List<Long> ids = sortedRowNumbers
-                .stream()
-                .map(SuitRowNumberUpdate::getId)
-                .collect(Collectors.toList());
+        suitDAO.save(Arrays.asList(
+                new Suit(null, "1", "1", new ArrayList<>(), 1, new HashSet<>(), 1),
+                new Suit(null, "2", "2", new ArrayList<>(), 2, new HashSet<>(), 2),
+                new Suit(null, "3", "3", new ArrayList<>(), 3, new HashSet<>(), 3)
+        ));
 
-        final List<Suit> actualSuits = suitDAO.findByIdIn(ids, new Sort(Sort.Direction.ASC, "id"));
-        Assert.assertEquals(expectedSuits, actualSuits);
+        final List<SuitRowNumberUpdate> numberUpdates = Arrays.asList(
+                getSuitRowNumberUpdateInstance(4L, 2),
+                getSuitRowNumberUpdateInstance(3L, 6),
+                getSuitRowNumberUpdateInstance(2L, 1),
+                getSuitRowNumberUpdateInstance(1L, 3)
+        );
 
-        final List<Suit> updatedSuits = updateByRowNumbers(actualSuits, sortedRowNumbers);
-        final List<Suit> suitsAfterChange = suitDAO.save(updatedSuits);
-        Assert.assertEquals(updatedSuits, suitsAfterChange);
-    }
+        final Map<Long, Integer> patch = numberUpdates
+                .stream()
+                .collect(Collectors.toMap(SuitRowNumberUpdate::getId, SuitRowNumberUpdate::getRowNumber));
 
-    private List<Suit> updateByRowNumbers(List<Suit> actualSuits, List<SuitRowNumberUpdate> sortedRowNumbers) {
-        for (int i = 0; i < actualSuits.size(); i++) {
-            final Integer newRowNumber = sortedRowNumbers.get(i).getRowNumber();
-            actualSuits.get(i).setRowNumber(newRowNumber);
-        }
-        return actualSuits;
-    }
+        final List<Suit> suits = suitDAO.findByIdInOrderById(patch.keySet());
 
-    private List<SuitRowNumberUpdate> getSortedSuitsRowNumbers(List<Suit> suits) {
-        List<Integer> rowNumbers = suits.stream().map(Suit::getRowNumber).collect(Collectors.toList());
-        Collections.shuffle(rowNumbers);
-
-        List<Suit> updateSuits = new ArrayList<>();
-        for (int i = 0; i < suits.size(); i++) {
-            Suit suit = suits.get(i);
-            updateSuits.add(new Suit(
-                    suit.getId(),
-                    suit.getName(),
-                    suit.getDescription(),
-                    suit.getCases(),
-                    suit.getPriority(),
-                    suit.getTags(),
-                    suit.getRowNumber())
-            );
-            updateSuits.get(i).setRowNumber(rowNumbers.get(i));
+        for (Suit suit : suits) {
```

Comment:
This code duplicates SuitService code. Tests should not duplicate any code. If someone changes SuitService code he'll have to modify testcase (which he does not about). Remove duplication.